### PR TITLE
get multicall for chain, export types

### PR DESCRIPTION
**Pull request by gsteenkamp89**
closes ACX-2585

- lets TSC handle bundling of ABI json files
- refactors multicall address maps
- refactors utilitY for connecting a signer to an instance of multicall for a chain.

### Breaking changes
- the interface for `getMulticall3` has changed somewhat and will affect consumers downstream

### CODE REVIEW

File: src/utils/Multicall.ts [hunk 1]
```diff
@@ -12,41 +13,49 @@ export type Call3 = {
...
+  CHAIN_IDs.BOBA,
+  CHAIN_IDs.LINEA,
+  CHAIN_IDs.LISK,
+  CHAIN_IDs.MAINNET,
+  CHAIN_IDs.MODE,
+  CHAIN_IDs.OPTIMISM,
+  CHAIN_IDs.POLYGON,
+  CHAIN_IDs.SCROLL,
+  CHAIN_IDs.ZORA,
+  CHAIN_IDs.REDSTONE,
```
Comment (pxrl) [inline src/utils/Multicall.ts:35]: Muh OCD!

Cool that Redstone has a deployment - I only checked the website and it doesn't list Redstone but I can see there is actually one deployed on chain.

```suggestion
const DETERMINISTIC_MULTICALL_CHAINS = [
  CHAIN_IDs.ARBITRUM,
  CHAIN_IDs.BASE,
  CHAIN_IDs.BLAST,
  CHAIN_IDs.BOBA,
  CHAIN_IDs.LINEA,
  CHAIN_IDs.LISK,
  CHAIN_IDs.MAINNET,
  CHAIN_IDs.MODE,
  CHAIN_IDs.OPTIMISM,
  CHAIN_IDs.POLYGON,
  CHAIN_IDs.REDSTONE,
  CHAIN_IDs.SCROLL,
  CHAIN_IDs.ZORA,
```

File: src/utils/abi/typechain/Multicall3.ts
Comment (pxrl) [inline src/utils/abi/typechain/Multicall3.ts:1]: Do we want to commit this or should we rely on it being produced during the build process?

File: src/utils/Multicall.ts (hunk 1 above)
Comment (gsteenkamp89) [reply src/utils/Multicall.ts:35]: you know, i didn't even notice these were in alphabetical order 🌭

**Review by dohaki — DISMISSED** [review]

**Review by dohaki — COMMENTED** [review]
Ah sorry, didn't mean to approve as CI still fails

**Comment by gsteenkamp89** [comment]
CI was failing because the files generated by typechain were not formatted after the codegen step.

**Review by dohaki — APPROVED** [review]
LGTM

**Pull request merged**